docs/book/src/providers/cluster-infrastructure.md

```diff
@@ -57,3 +57,32 @@ The following diagram shows the typical logic for a cluster infrastructure provi
     1. If any errors are encountered, exit the reconciliation
 1. Remove the provider-specific finalizer from the resource
 1. Patch the resource to persist changes
+
+## RBAC
+
+### Provider controller
+
+A cluster infrastructure provider must have RBAC permissions for the types it defines. If you are using `kubebuilder` to
+generate new API types, these permissions should be configured for you automatically. For example, the AWS provider has
+the following configuration for its `AWSCluster` type:
+
+```
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awsclusters,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awsclusters/status,verbs=get;update;patch
+```
+
+A cluster infrastructure provider may also need RBAC permissions for other types, such as `Cluster`. If you need
+read-only access, you can limit the permissions to `get`, `list`, and `watch`. The AWS provider has the following
+configuration for retrieving `Cluster` resources:
+
+```
+// +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;clusters/status,verbs=get;list;watch
+```
+
+### Cluster API controllers
+
+The Cluster API controller for `Cluster` resources is configured with full read/write RBAC permissions for all resources
```

Comment:
It might be good to mention that the write permission is for adding an ownerRef and/or labels only and not for general mutation of the resources.

Reply:
Added a new para for this - PTAL